spec.md

```diff
@@ -5,4599 +5,420 @@ keywords: registry, on-prem, images, tags, repository, distribution, api, advanc
-
-```json
-{
-   "annotations": {
-      "com.example.key1": "value1",
-      "com.example.key2": "value2"
-   },
-   "config": {
-      "digest": "sha256:6f4e69a5ff18d92e7315e3ee31c62165ebf25bfa05cad05c0d09d8f412dae401",
-      "mediaType": "application/vnd.oci.image.config.v1+json",
-      "size": 452
-   },
-   "layers": [
-      {
-         "digest": "sha256:6f4e69a5ff18d92e7315e3ee31c62165ebf25bfa05cad05c0d09d8f412dae401",
-         "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
-         "size": 78343
-      }
-   ],
-   "schemaVersion": 2
-}
+Location: <blob-location>
 ```
 
-The client SHOULD verify the returned manifest signature for authenticity before fetching layers.
-
-##### Existing Manifests
-
-The image manifest can be checked for existence with the following url:
-
-```HTTP
-HEAD /v2/<name>/manifests/<reference>
-```
+With `<blob-location>` being a pullable blob URL.
 
-The `name` and `reference` parameter identify the image and are REQUIRED.
-The reference MAY include a tag or digest.
+---
 
-A `404 Not Found` response will be returned if the image is unknown to the registry.
-If the image exists and the response is successful the response will be as follows:
+To push a blob monolithically by using a POST request followed by a PUT request, there are two steps:
+1. Obtain a session id (upload URL)
+2. Upload the blob to said URL
 
-```HTTP
-200 OK
-Content-Length: <length of manifest>
-Docker-Content-Digest: <digest>
-```
+To obtain a session ID, perform a `POST` request to a URL in the following format:
 
-#### Pulling a Layer
+[4a](#Endpoints) `/v2/<name>/blobs/uploads/`
 
-Layers are stored in the blob portion of the registry, keyed by digest.
-Pulling a layer is carried out by a standard http request.
-The URL is as follows:
+Here, `<name>` refers to the namespace of the repository. Upon success, the response MUST have a code of `202 Accepted`, and MUST include the following header:
 
-```HTTP
-GET /v2/<name>/blobs/<digest>
 ```
-
-Access to a layer will be gated by the `name` of the repository but is identified uniquely in the registry by `digest`.
-
-This endpoint MAY issue a 307 (302 for < HTTP 1.1) redirect to another service for downloading the layer and clients SHOULD be prepared to handle redirects.
-
-This endpoint SHOULD support aggressive HTTP caching for image layers.
-Support for Etags, modification dates and other cache control headers SHOULD be included.
-To allow for incremental downloads, `Range` requests SHOULD be supported, as well.
-
-### Pushing An Image
-
-Pushing an image works in the opposite order as a pull.
-After assembling the image manifest, the client MUST first push the individual layers.
-When the layers are fully pushed into the registry, the client SHOULD upload the signed manifest.
-
-The details of each step of the process are covered in the following sections.
-
-#### Pushing a Layer
-
-All layer uploads use two steps to manage the upload process.
```

Comment:
..

Reply:
Strictly speaking, a layer upload can be done in a single step: a POST request containing the blob with the ?digest=<digest> query parameter will complete an upload in a single-shot fashion.